### PR TITLE
Feat/invt cpc 897 update product front end

### PR DESCRIPTION
**Pull request by AlexeiTimbro**
JIRA: link to jira ticket
https://champlainsaintlambert.atlassian.net/browse/CPC-897

What is the ticket about and why are we making this change?

This ticket is about the completion of the front end for the UpdateProductInInventory. This was necessary because an Admin of the Pet Clinic needs to be able to update/edit a product inside of an inventory if they made a mistake when creating a product for example or they want to change a field inside the product like the quantity. 

What are the various changes and what other modules do those changes affect?

- Front-end Edit button fully working
- Made a new Form for the Update only
- Made a new folder called product-update-form where I added:
1. Controller
2. Component
3. .js file
4. html template form

- Added a test that tests one of the exceptions in the backend as well to get 100% jacoco coverage for the Update method

## Before and After UI (Required for UI-impacting PRs)

Before Clicking edit: (Look at the first product's Values)
image

After clicking the edit button for the first product: (Pre-filled fields)
image

Changing the values:
image

After clicking submit: (Fields are updated)
image

### CODE REVIEW

**Review by SandwichKing978 — APPROVED** [review]
Great job on making the update product work, I know you put a lot of effort in it keep up the good work.

File: inventory-service/src/main/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceImpl.java [hunk 1]
```diff
@@ -92,17 +92,18 @@ public Mono<ProductResponseDTO> updateProductInInventory(Mono<ProductRequestDTO>
...
                             } else {
                                 return productRepository.findProductByProductId(productId)
                                         .flatMap(existingProduct -> {
-                                            Product updatedProduct = EntityDTOUtil.toProductEntity(requestDTO);
-                                            updatedProduct.setProductId(existingProduct.getProductId());
-                                            updatedProduct.setInventoryId(existingProduct.getInventoryId());
+                                            existingProduct.setProductName(requestDTO.getProductName());
+                                            existingProduct.setProductDescription(requestDTO.getProductDescription());
+                                            existingProduct.setProductPrice(requestDTO.getProductPrice());
+                                            existingProduct.setProductQuantity(requestDTO.getProductQuantity());
```
Comment (Iantomasi) [inline inventory-service/src/main/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceImpl.java:105]: Smart changes my padawan. This method tweak seems to be what rendered your method successful.

**Review by Iantomasi — APPROVED** [review]
Good stuff alexei. I know this pull request took a lot of time and effort

File: inventory-service/src/main/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceImpl.java (hunk 1 above)
Comment (AlexeiTimbro) [reply inventory-service/src/main/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceImpl.java:105]: Yes, indeed it was.

**Pull request merged**